addresses-scheme/readme.md

```diff
@@ -0,0 +1,134 @@
+
+### The Clean-Namespaces Goal: Avoid polluting the scheme namespace with multiple schemes
+
+If we do this wrong, the growth of decentralized web technologies will cause a proliferation of schemes that will quickly become unwieldy, will discourage interoperability, and will maintain a high barrier to innovation in the protocol space.
+
+## The `dweb:` Approach
+
+### Strengths of this Approach
+
+_PLEASE HELP FILL THESE_
+
+#### Strength: Prevents Proliferation of Schemes and Namespaces
+
+NOT doing this now effectively shuts down future possibilities by making new namespaces "not worth doing the work of introducing a URL scheme".
+
+#### Strength: Getting away from the `://`
+as @lidel [comments](https://github.com/ipfs/specs/pull/153#discussion_r104291285)
+> I kinda like this _aesthetic_. No matter what prefix is picked, it looks better than anything with `://`
+> `/webfs/ipfs/QmT272yei1Zn1eAUq5P9nZyeaKP4oJmVv7CbYvUPyk3aLj/hobby.jpg`
+> `/dweb/ipfs/QmT272yei1Zn1eAUq5P9nZyeaKP4oJmVv7CbYvUPyk3aLj/hobby.jpg`
+> `/x/ipfs/QmT272yei1Zn1eAUq5P9nZyeaKP4oJmVv7CbYvUPyk3aLj/hobby.jpg`
+
+
+### Criticisms of this Approach
+_PLEASE HELP FILL THESE_
+
+#### Criticism: It's not clear how this helps the whole ecosystem
+@lidel [comments](https://github.com/ipfs/specs/pull/152#discussion_r104291969)
+> I feel it should be explicitly stated in the document if this single scheme aims to be for IPFS ecosystem only, or something that can be adopted by other distributed systems.
+>
+> If we want to share it with others, eg. Ethereum, Tor Onion Services etc, then I see potential problem with single scheme: who is responsible for muxing when multiple vendors are in play?
+Let's say we have: `dweb://ipfs/Qmbar..` and `dweb://foo/buz`.  If IPFS browser add-on provides handler for `dweb://`, FOO is unable to handle its own URIs.
+We should plan for this now, and have contingency written down..
+
+#### Criticism: Doesn't fit with the way Browsers identify origins
+
+Some current browsers have trouble with the construction: origin = a:/b/c because they expect: `origin = a:/b`, even though that's not in the URI spec.
+
+### Designing the `dweb:` Schema
+
+A draft spec for the `dweb:` schema is under way at https://github.com/ipfs/in-web-browsers/issues/28
+
+## The `ipfs:` Approach
+
+The short-term fix that people reach for is to create an `ipfs:` schema, as proposed in https://github.com/ipfs/specs/pull/139. That approach seems simple at first, but it's got problems.
+
+### Strengths of this Approach
+
+#### Strength: Fits with Established Conventions & Expectations around URLs
+
+#### Strength: Fits with the way browsers identify origins
+
+When enforcing the Single Origin Policy, some current browsers expect: `origin = a:/b`, even though that's not what the URI spec calls for. The `ipfs:` approach fits with that expectation, with URIs like `ipfs:/<root-hash>/foo/bar`, the origin would be `ipfs:/<root-hash>`
```

Comment:
Interesting point. 
A couple of things I will have to update the doc to reflect:

1. @pawal's spec at https://github.com/ipfs/specs/pull/139 includes a brief reference to the `fs:` scheme which is fleshed out in @lgierth's https://github.com/ipfs/specs/tree/doc/fs-paths/fs-paths 
2. A couple of points are implicit in @jbenet's arguments, which hang off the fact that an `ipfs:` scheme should, due to its name, only return ipfs content: 
    * if you create an `ipfs:` scheme, the `/ipfs/` in the path is redundant. 
    * if you create an `ipfs:` scheme you will end up needing more schemes: `ipns:`, `ipld:`, etc.